### ISSUE TITLE
Class fields do not produce forbidden semi-colon errors

### ISSUE THREAD

**Issue by justinmchase** [eslint/eslint#6236]
**What version of ESLint are you using?**
2.10.2

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLint

**Please show your full configuration:**
```
{
    "parser": "babel-eslint",
    "extends": "standard",
    "parserOptions": {
      "sourceType": "module"
    },
    "env": {
      "node": true,
      "browser": true,
      "mocha": true,
      "es6": true
    },
    "plugins": [
      "standard"
    ]
}
```

**What did you do? Please include the actual source code causing the issue.**

Standard forbids semi-colon usage
```
class Foo {
  bar = 'example'; // should be an error
}
```

**What did you expect to happen?**
I expected the linter to product an error about the semi-colon on the member field.

**What actually happened? Please include the actual, raw output from ESLint.**
No error, no output.

**Comment by kaicataldo** [comment]
Hey, thanks for the issue. It looks like you're using babel-eslint, which uses its own parser. I just tried this using http://astexplorer.net/. It looks like Espree (the default ESLint parser) does actually throw on the semicolon, while Babylon (babel-eslint's parser) does not. Can you try this with the default parser (you can do so by deleting the parser key from your .eslintrc), and let us know if it's happening with the default parser as well? If not, this issue should be filed with babel-eslint.

**Comment by ilyavolodin** [comment]
Class Property Declarations are stage 1 proposal (so, very far from being accepted). As such, ESLint doesn't have support for them. You should open an issue for `eslint-plugin-babel`. Closing, as this is not supported.

**Comment by justinmchase** [comment]
Ok, thanks for the reply. I'll push the bug on up.

**Comment by hzoo** [comment]
This is basically https://github.com/babel/eslint-plugin-babel/issues/43 - modify semi and friends to support class properties.
